upstairs/src/volume.rs

```diff
@@ -611,6 +611,36 @@ impl Volume {
 
         Ok(())
     }
+
+    pub async fn volume_extent_info(
+        &self,
+    ) -> Result<VolumeInfo, CrucibleError> {
+        // A volume has multiple levels of extent info, not just one.
+        let mut volumes = Vec::new();
+
+        for sub_volume in &self.sub_volumes {
+            match sub_volume.query_extent_info().await? {
+                Some(ei) => {
+                    assert_eq!(self.block_size, ei.block_size);
```

Comment:
Add a comment here as to why this can never happen (is it checked on Volume construction?)